### PR TITLE
feat: add C `ndarray` implementation for `blas/base/dsdot`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request adds C `ndarray` API for `blas/base/dsdot`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/dsdot/README.md [hunk 1]
```diff
@@ -227,6 +227,32 @@ The function accepts the following arguments:
 double c_dsdot( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const float *Y, const CBLAS_INT strideY );
 ```
 
+#### c_dsdot_ndarray( N, \*X, strideX, offsetX, \*Y, strideY, offsetY )
+
+Computes the dot product of two single-precision floating-point vectors with extended accumulation and result using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/README.md]:
```suggestion
Computes the dot product of two single-precision floating-point vectors with extended accumulation and result and using alternative indexing semantics.
```

Updated to match the JS description above.

File: lib/node_modules/@stdlib/blas/base/dsdot/README.md [hunk 2]
```diff
@@ -227,6 +227,32 @@ The function accepts the following arguments:
...
+-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
+-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.
+-   **Y**: `[in] float*` second input array.
+-   **strideY**: `[in] CBLAS_INT` index increment for `Y`.
+-   **offsetY**: `[in] CBLAS_INT` starting index for `Y`.
+
+```c
+double c_dsdot_ndarray( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX, const float *Y, const CBLAS_INT strideY, const CBLAS_INT offsetY );
+```
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/README.md:255]: @aman-095 You updated `example.c`, but not the example below. Was that intentional?

File: lib/node_modules/@stdlib/blas/base/dsdot/include/stdlib/blas/base/dsdot.h [hunk 3]
```diff
@@ -36,6 +36,11 @@ extern "C" {
 */
 double API_SUFFIX(c_dsdot)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const float *Y, const CBLAS_INT strideY );
 
+/**
+* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/include/stdlib/blas/base/dsdot.h]:
```suggestion
* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result and using alternative indexing semantics.
```

File: lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_cblas.c [hunk 4]
```diff
@@ -28,8 +29,26 @@
...
 * @param strideY  Y stride length
-* @return         the dot product
+* @return         dot product
 */
 double API_SUFFIX(c_dsdot)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const float *Y, const CBLAS_INT strideY ) {
 	return API_SUFFIX(cblas_dsdot)( N, X, strideX, Y, strideY );
 }
+
+/**
+* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_cblas.c]:
```suggestion
* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result and using alternative indexing semantics.
```

File: lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_f.c [hunk 5]
```diff
@@ -28,10 +29,31 @@
...
+* @return         dot product
 */
 double API_SUFFIX(c_dsdot)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const float *Y, const CBLAS_INT strideY ) {
 	double dot;
 	dsdotsub( &N, X, &strideX, Y, &strideY, &dot );
 	return dot;
 }
+
+/**
+* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_f.c]:
```suggestion
* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result and using alternative indexing semantics.
```

File: lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_ndarray.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/blas/base/dsdot.h"
+#include "stdlib/blas/base/shared.h"
+
+static const CBLAS_INT M = 5;
+
+/**
+* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_ndarray.c]:
```suggestion
* Computes the dot product of two single-precision floating-point vectors with extended accumulation and result and using alternative indexing semantics.
```

File: lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_ndarray.c [hunk 7]
```diff
@@ -0,0 +1,79 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dsdot/src/dsdot_ndarray.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

**Review by kgryte — COMMENTED** [review]
Apart from the question about the C example in the README, this PR LGTM.

File: lib/node_modules/@stdlib/blas/base/dsdot/README.md (hunk 2 above)
Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/dsdot/README.md:255]: No, @kgryte it was not intentional, I think I missed that. Updated.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Pull request merged**